src/OpenTelemetry.Extensions.PersistentStorage/FileBlobProvider.cs

```diff
@@ -115,31 +116,37 @@ public IEnumerable<IPersistentBlob> GetBlobs()
             }
         }
 
-        /// <inheritdoc/>
-        public IPersistentBlob GetBlob()
+        protected override bool OnTryCreateBlob(byte[] buffer, int leasePeriodMilliseconds, [NotNullWhen(true)] out PersistentBlob blob)
         {
-            return this.GetBlobs().FirstOrDefault();
+            blob = this.CreateFileBlob(buffer, leasePeriodMilliseconds);
+            if (blob == null)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
nit: consider simplifying this:

```suggestion
            return blob != null;
```

Reply:
This class has two more `if` blocks that can be simplified in the same way. GitHub wouldn't let me give suggestions for the others.